analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/DeliveringDebugFeaturesInProduction.cs

```diff
@@ -42,5 +42,20 @@ public sealed class DeliveringDebugFeaturesInProduction : DeliveringDebugFeature
         private bool IsDevelopmentCheck(SyntaxNode node, SemanticModel semanticModel) =>
             node is InvocationExpressionSyntax condition
             && IsValidationMethod(semanticModel, condition, condition.Expression.GetIdentifier()?.ValueText);
+
+        private static SyntaxNode EnclosingScope(SyntaxNode node) =>
+            node.Ancestors().FirstOrDefault(x => x.IsAnyKind(
+                SyntaxKind.AddAccessorDeclaration,
+                SyntaxKind.ConstructorDeclaration,
+                SyntaxKind.DestructorDeclaration,
+                SyntaxKind.GetAccessorDeclaration,
+                SyntaxKind.GlobalStatement,
+                SyntaxKindEx.InitAccessorDeclaration,
+                SyntaxKindEx.LocalFunctionStatement,
+                SyntaxKind.MethodDeclaration,
+                SyntaxKind.ParenthesizedLambdaExpression,
+                SyntaxKind.RemoveAccessorDeclaration,
+                SyntaxKind.SetAccessorDeclaration,
+                SyntaxKind.SimpleLambdaExpression));
```

Comment:
Initializer blocks are missing (SyntaxKind.EqualsValueClause)
```cs
public class C
{
    int i = 0;
    int I { get; } = 0;
}
```

Reply:
And three more occasions where we have expressions that do not belong to a method directly:
* [ConstructorInitializerSyntax](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntax.constructorinitializersyntax?view=roslyn-dotnet-4.7.0)
* [PrimaryConstructorBaseTypeSyntax](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntax.primaryconstructorbasetypesyntax?view=roslyn-dotnet-4.7.0)
 
[sharplab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGED2AbbEDGMAlpgHYAyRMECU2yANDCFMjAD4ACAzAAScAmXgBEAsACgA3hN6z+fYQAoipGLyIBKXpN4BfCfvESe/IehAiJ08XPm90ijRZgALIskUBGXgGpenrQB6QP9fUPd1UnUYAHJkXkwAdyj8MmoADxgGXmo2XjBMBF4AQQQAczgAWwhVAGUAT1UodIA6AFF0gAckZGQSKMSqF14ABSgkVRaxibUIrFI2BDhCQoBJUioiOiIALxoGpvTeGTkdQ1sTBxVZp15gVghlHx8gwKJn9XiVSJh4pJS0hBMidZGcDBJjHwkKkEAATXgAJWUql4qw0AG4IeITNDCvDasi1GiLEjVs8MbJgmS/BEyNE4glkrxUqogTAgA)